src/receive_imf.rs

```diff
@@ -2324,6 +2329,82 @@ async fn handle_edit_delete(
     Ok(())
 }
 
+async fn handle_post_message(
+    context: &Context,
+    mime_parser: &MimeMessage,
+    from_id: ContactId,
+) -> Result<()> {
+    if let Some(mimeparser::PreMessageMode::PostMessage) = &mime_parser.pre_message {
+        // if Pre-Message exist, replace attachment
+        // only replacing attachment ensures that doesn't overwrite the text if it was edited before.
+        let rfc724_mid = mime_parser
+            .get_rfc724_mid()
+            .context("expected Post-Message to have a message id")?;
+
+        let Some(msg_id) = message::rfc724_mid_exists(context, &rfc724_mid).await? else {
+            warn!(
+                context,
+                "Download Post-Message: Database entry does not exist."
```

Comment:
```suggestion
                "Download Post-Message {rfc724_mid}: Database entry does not exist."
```
We have multi-transport now, so `receive_imf_inner()` can be run in parallel AFAIU and logging some context may help. But overall we've been having this "logging context" problem for a long time already, would be nice to solve it in some generic way